crates/transpiler/src/passes/basis_translator/mod.rs

```diff
@@ -47,80 +52,93 @@ use crate::equivalence::EquivalenceLibrary;
 use crate::target::Qargs;
 use crate::target::QargsRef;
 use crate::target::Target;
-use crate::TranspilerError;
 
-type InstMap = IndexMap<GateIdentifier, BasisTransformOut, ahash::RandomState>;
-type ExtraInstructionMap<'a> = IndexMap<&'a PhysicalQargs, InstMap, ahash::RandomState>;
+type AhashIndexMap<K, V> = IndexMap<K, V, ahash::RandomState>;
+type AhashIndexSet<O> = IndexSet<O, ahash::RandomState>;
+type InstMap = AhashIndexMap<GateIdentifier, BasisTransformOut>;
+type ExtraInstructionMap<'a> = AhashIndexMap<&'a PhysicalQargs, InstMap>;
 type PhysicalQargs = SmallVec<[PhysicalQubit; 2]>;
 
-#[allow(clippy::too_many_arguments)]
-#[pyfunction(name = "base_run", signature = (dag, equiv_lib, qargs_with_non_global_operation, min_qubits, target_basis=None, target=None, non_global_operations=None))]
-pub fn run_basis_translator(
-    py: Python<'_>,
-    dag: &DAGCircuit,
+#[pyfunction(name = "base_run", signature = (dag, equiv_lib, min_qubits, target=None, target_basis=None))]
+fn py_run_basis_translator(
+    dag: DAGCircuit,
     equiv_lib: &mut EquivalenceLibrary,
-    qargs_with_non_global_operation: HashMap<Qargs, HashSet<String>>,
     min_qubits: usize,
+    target: Option<&mut Target>,
     target_basis: Option<HashSet<String>>,
-    target: Option<&Target>,
-    non_global_operations: Option<HashSet<String>>,
 ) -> PyResult<Option<DAGCircuit>> {
+    run_basis_translator(dag, equiv_lib, min_qubits, target, target_basis)
+        .map_err(|e| e.to_py_err())
+}
+
+pub fn run_basis_translator(
+    dag: DAGCircuit,
```

Comment:
Why does this take an owned `DAGCircuit`? The basis translator shouldn't need to delete the dagcircuit it receives. I'd expect this to either be `&mut DAGCircuit` if it mutates the dag in place, or `&DAGCircuit` if it only reads the dag and returns a copy.

Reply:
Fixed in ec026b9